xarray/core/common.py

```diff
@@ -343,17 +343,21 @@ def groupby(self, group, squeeze=True):
             group = self[group]
         return self.groupby_cls(self, group, squeeze=squeeze)
 
-    def rolling(self, min_periods=None, center=False, **kwarg):
-        """Returns a Rolling object for performing moving window operations.
+    def rolling(self, min_periods=None, center=False, **windows):
+        """
+        Moving window object.
 
         Parameters
         ----------
+        obj : Dataset or DataArray
+            Object to window.
         min_periods : int, default None
             Minimum number of observations in window required to have a value
-            (otherwise result is NA).
+            (otherwise result is NA). The default, None, is equivalent to
+            setting min_periods equal to the size of the window.
```

Comment:
hmmm, you must not have bottleneck in your environment because I get:

```Python
In [6]: d.rolling(x=2).mean()
Out[6]: 
<xarray.DataArray (x: 11)>
array([ nan,  nan,  nan,  1.5,  nan,  nan,  3.5,  4.5,  nan,  nan,  6.5])
Coordinates:
  * x        (x) int64 0 1 2 3 4 5 6 7 8 9 10
```

I suppose we should open an issue on this. I guess we need to use the non-nan safe numpy methods in `Rolling.reduce` to get the same behavior. We'll have to come up with a solution to get this to work in a vectorized manor.